### PR TITLE
Add Google Flatbuffers recipe

### PR DESCRIPTION
**Pull request by wesm**

### CODE REVIEW

**Comment by wesm** [comment]
@msarahan this recipe should also build on Windows without a hitch; someone more experienced will need to write the build script though

File: recipes/flatbuffers/build.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  export CXXFLAGS="${CXXFLAGS} -stdlib=libc++ -std=c++11"
+  export LDFLAGS="${LDFLAGS} -mmacosx-version-min=${MACOSX_VERSION_MIN}"
+  export LDFLAGS="${LDFLAGS} -stdlib=libc++ -std=c++11"
+  export LINKFLAGS="${LDFLAGS}"
+  export MACOSX_DEPLOYMENT_TARGET=10.7
+fi
+
+cmake -DCMAKE_INSTALL_PREFIX:PATH=$PREFIX -DFLATBUFFERS_BUILD_TESTS=OFF .
+make
+make install
```
Comment (msarahan) [inline recipes/flatbuffers/build.sh]: Is there a make check or equivalent?

**Comment by msarahan** [comment]
Please add me as a maintainer, as I'm going to do the Windows work.  Otherwise, please add the make check or equivalent, and this is good to merge for now - and I can add the windows support on the feedstock later.

**Comment by wesm** [comment]
@msarahan you now have push on my fork

File: recipes/flatbuffers/build.sh (hunk 1 above)
Comment (wesm) [inline recipes/flatbuffers/build.sh]: I'm not sure. Probably

Comment (jakirkham) [inline recipes/flatbuffers/build.sh]: What @msarahan means is we like to run the code's test suite during the build process. Generally, we don't expect to find bugs in the code (though sometimes that happens), but in our build by doing this. It makes sure that we have exercised the package fairly well and can be relatively confident that the binary we ship works in most cases. If we can add a `make check`, `ctest`, or similar step here, that would be great.

File: recipes/flatbuffers/build.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  export MACOSX_VERSION_MIN="10.7"
+  export CXXFLAGS="${CXXFLAGS} -mmacosx-version-min=${MACOSX_VERSION_MIN}"
+  export CXXFLAGS="${CXXFLAGS} -stdlib=libc++ -std=c++11"
+  export LDFLAGS="${LDFLAGS} -mmacosx-version-min=${MACOSX_VERSION_MIN}"
+  export LDFLAGS="${LDFLAGS} -stdlib=libc++ -std=c++11"
+  export LINKFLAGS="${LDFLAGS}"
+  export MACOSX_DEPLOYMENT_TARGET=10.7
+fi
+
+cmake -DCMAKE_INSTALL_PREFIX:PATH=$PREFIX -DFLATBUFFERS_BUILD_TESTS=OFF .
```
Comment (jakirkham) [inline recipes/flatbuffers/build.sh]: I'm guessing we will need to set `FLATBUFFERS_BUILD_TESTS` to `ON` to run the test suite below.

File: recipes/flatbuffers/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  skip: true  # [win]
+  number: 0
+
+requirements:
+  build:
+    - cmake
+
+test:
+  commands:
+    - test -f $PREFIX/bin/flatc
```
Comment (jakirkham) [inline recipes/flatbuffers/meta.yaml]: Is there a help or version command that we can run with `flatc`? While running such a simple command like help or version is not a great test, it is good at attempt to load up any relevant dynamic libraries. If this fails, we know we need to go back and handle linking errors. If not, we can be pretty confident it linked together ok.

Comment (wesm) [inline recipes/flatbuffers/meta.yaml]: We can build and run the unit tests.

**Comment by msarahan** [comment]
@wesm regarding push access: what I was really asking for was just adding my name to the recipe-maintainers list in meta.yaml.  This list is not just informational - it is used to create the github permissions on the feedstock when this staged-recipes PR is merged.  Push access to your fork is unnecessary, and I personally would prefer to just submit PR's against your fork, or once this PR is merged, against the feedstock repo.

**Comment by wesm** [comment]
@msarahan ok I think this can be merged. I am not sure what the appveyor failure is about

**Comment by ocefpaf** [comment]
@wesm the AppVeyor failure is b/c I cancelled the build. You have `skip: true  # [win]`, so that has no bearing here and the build is OK. We usually cancel to avoid getting other projects backlogged. (AppVeyor is quite slow.)

PS: We recommend people to add `[skip appveyor]` to the commit messages when skipping Windows builds. That way AppVeyor will not start.

**Comment by wesm** [comment]
Noted, thanks

**Comment by ocefpaf** [comment]
@wesm I believe @msarahan "shift" is over.  I am merging this and you guys can iterate further on the feedstocks. Thanks for the contribution!

**Pull request merged**